File: tests/07_phases_ompss.dg/fortran/success_task_23.f90
! <testinfo>
! test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
! </testinfo>
PROGRAM P
    IMPLICIT NONE
    INTEGER, ALLOCATABLE :: V(:, :)

    ALLOCATE(V(10, 5))

    V = 1
    !$OMP TARGET DEVICE(SMP) COPY_DEPS
    !$OMP TASK INOUT(V)
        V(1, :)= -2
    !$OMP END TASK

    !$OMP TASKWAIT
    if (ANY(V(1,:) /= -2)) STOP 1
    if (ANY(V(2:,:) /= 1)) STOP 2
END PROGRAM P

File: tests/07_phases_ompss.dg/c/success_task_23.c
/*
<testinfo>
test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
test_CFLAGS=-std=gnu99
test_nolink=yes
</testinfo>
*/
void dealloc_tiled_matrix(int MBS, int NBS, int M, int N, double (*a)[N/NBS][MBS][NBS]) {
        // Build a fictitious dependency structure to free the whole tiled matrix at once
        for (int i=0; i<M/MBS; i ++) {
                for (int j=0; j<N/NBS; j++) {
                        if (i != 0 || j != 0) {
                                #pragma omp task inout(a[i][j]) concurrent(a[0][0])
                                {
                                }
                        }
                }
        }

        //#pragma omp task inout(a[0][0])
        //free(a);
}


int main(int argc, char *argv[])
{
    return 0;
}
